api/src/wfl/module/covid.clj

```diff
@@ -17,6 +19,57 @@
 (defn ^:private get-snapshots-from-workspace
   [workspace])
 
+(defn throw-unless-column-exists [dataset table column-name]
+  "Throw or return the column from `table` with `column-name`"
+  (let [[result & more :as all] (-> table
+                                    (get-in [:columns])
+                                    (->> (filter (comp #{column-name} :name))))]
+    (when-not result
+      (throw (ex-info "No column with name" {:name column-name :dataset dataset})))
+    result))
+
+(defn throw-unless-table-exists [dataset table-name column-name]
+  "Throw or return the table from `dataset` with `table-name`"
+  (let [[result & more :as all] (-> dataset
+                                    (get-in [:schema :tables])
+                                    (->> (filter (comp #{table-name} :name))))]
+    (when-not result
+      (throw (ex-info "No table with name" {:name table-name :dataset dataset})))
+    (when result
+      (throw-unless-column-exists dataset result column-name))
+    result))
+
+(defn verify-source!
+  "Verify that the `dataset` exists and that the WFL has the necessary permissions to read it"
+  [{:keys [name dataset] :as source}]
+  (when-not (= (:name source) "Terra DataRepo")
+    (throw (ex-info "Unknown Source" {:source source})))
+  (try
+    (-> (datarepo/dataset (:dataset source))
+        (throw-unless-table-exists "flowcell" "updated"))
```

Comment:
Are these hardcoded values available in the source request?

Reply:
These values were removed due to another comment